src/EditorFeatures/Test/CodeGeneration/AbstractCodeGenerationTests.cs

```diff
@@ -14,31 +17,99 @@ namespace Microsoft.CodeAnalysis.Editor.UnitTests.CodeGeneration
     [UseExportProvider]
     public abstract class AbstractCodeGenerationTests
     {
+        private static SyntaxNode Simplify(
+            AdhocWorkspace workspace,
+            SyntaxNode syntaxNode,
+            string languageName)
+        {
+            var projectId = ProjectId.CreateNewId();
+
+            var project = workspace.CurrentSolution
+                .AddProject(projectId, languageName, $"{languageName}.dll", languageName).GetProject(projectId);
+
+            var normalizedSyntax = syntaxNode.NormalizeWhitespace().ToFullString();
+            var document = project.AddMetadataReference(TestReferences.NetFx.v4_0_30319.mscorlib)
+                .AddDocument("Fake Document", SourceText.From(normalizedSyntax));
+
+            var annotatedDocument = document.WithSyntaxRoot(
+                    document.GetSyntaxRootAsync().Result.WithAdditionalAnnotations(Simplification.Simplifier.Annotation));
+
+            var annotatedRootNode = annotatedDocument.GetSyntaxRootAsync().Result;
+
+            var simplifiedDocument = Simplification.Simplifier.ReduceAsync(annotatedDocument).Result;
+
+            var rootNode = simplifiedDocument.GetSyntaxRootAsync().Result;
+
+            return rootNode;
+        }
+
+        private static SyntaxNode WrapExpressionInBoilerplate(SyntaxNode expression, SyntaxGenerator codeDefFactory)
+        {
+            return codeDefFactory.CompilationUnit(
+                codeDefFactory.NamespaceImportDeclaration(codeDefFactory.IdentifierName("System")),
+                codeDefFactory.ClassDeclaration(
+                    "C",
+                    members: new[]
+                    {
+                        codeDefFactory.MethodDeclaration(
+                            "Dummy",
+                            returnType: null,
+                            statements: new[]
+                            {
+                                codeDefFactory.LocalDeclarationStatement("test", expression)
+                            })
+                    })
+                );
+        }
+
         internal void Test(
             Func<SyntaxGenerator, SyntaxNode> nodeCreator,
-            string cs, string vb)
+            string cs, string csSimple,
+            string vb, string vbSimple)
         {
             var hostServices = MefV1HostServices.Create(TestExportProvider.ExportProviderWithCSharpAndVisualBasic.AsExportProvider());
             var workspace = new AdhocWorkspace(hostServices);
 
-            if (cs != null)
```

Comment:
might be a good idea to assert that at least one of those 4 tests is not null.  We've def had cases in the past where you could call a test method with not enough info and always pass :D

Reply:
Good idea, I'll add that :)